node/src/node.rs

```diff
@@ -13,8 +15,93 @@ pub struct Node<N: GossipNetwork, S: Storage, R: RawRepository> {
 }
 
 impl SimperbyNode {
-    pub async fn initialize(_config: &Config) -> Result<Self> {
-        todo!()
+    pub async fn initialize(config: Config, path: &str) -> Result<Self> {
+        // Step 0: initialize the repository module
+        let peers: Vec<Peer> = serde_json::from_str(
+            &tokio::fs::read_to_string(&format!("{}/peers.json", path)).await?,
+        )?;
+        let peers = SharedKnownPeers::new_static(peers.clone());
+        let raw_repository = RawRepositoryImpl::open(&format!("{}/repository/repo", path)).await?;
+        let repository = DistributedRepository::new(
+            raw_repository,
+            simperby_repository::Config {
+                mirrors: config.public_repo_url.clone(),
+                long_range_attack_distance: 3,
+            },
+            peers.clone(),
+        )
+        .await?;
+
+        // Step 1: initialize configs
+        let last_finalized_header = repository.get_last_finalized_block_header().await?;
+        let reserved_state = repository.get_reserved_state().await?;
+        let governance_dms_key = simperby_governance::generate_dms_key(&last_finalized_header);
+        let consensus_dms_key = simperby_consensus::generate_dms_key(&last_finalized_header);
+        let network_config = NetworkConfig {
+            network_id: reserved_state.genesis_info.chain_name.clone(),
+            ports: vec![
+                (governance_dms_key.clone(), 1555),
+                (consensus_dms_key.clone(), 1556),
+            ]
+            .into_iter()
+            .collect(),
+            members: reserved_state
+                .members
+                .iter()
+                .map(|m| m.public_key.clone())
+                .collect(),
+            public_key: config.public_key.clone(),
+            private_key: config.private_key.clone(),
+        };
+        let dms_config = dms::Config {
+            fetch_interval: Some(std::time::Duration::from_millis(500)),
+            broadcast_interval: Some(std::time::Duration::from_millis(500)),
+            network_config,
+        };
+
+        // Step 2: initialize the governance module
+        StorageImpl::create(&format!("{}/governance/dms", path))
+            .await
+            .unwrap();
+        let storage = StorageImpl::open(path).await.unwrap();
+        let dms = Dms::new(
+            storage,
+            governance_dms_key,
+            dms_config.clone(),
+            peers.clone(),
+        )
+        .await?;
+        let governance = Governance::new(dms, Some(config.private_key.clone())).await?;
+
+        // Step 3: initialize the consensus module
+        StorageImpl::create(&format!("{}/consensus/dms", path))
+            .await
+            .unwrap();
+        let storage = StorageImpl::open(path).await.unwrap();
```

Comment:
It should be
`let storage = StorageImpl::open(&format!("{}/consensus/dms", path)).await.unwrap();`